### ISSUE TITLE
Restructuring suggestion

### ISSUE THREAD

**Issue by dougajmcdonald** [gareththegeek/corewar#4]
As we talked about, I think this would do well from a quick chop up.

I'd suggest we look at initially splitting into a slightly different folder structure but maybe investigate `git submodules`.

I'd suggest something like:
```
- corewars
-- standards
-- warriors
-- parser
--- tests
-- simulator
--- tests
```

thoughts?

**Comment by gareththegeek** [comment]
I concur

**Comment by dougajmcdonald** [comment]
I think initially the UI and entry point could site in `corewars` but maybe could be split out into a `ui` / `web` kind of folder if needed

**Comment by gareththegeek** [comment]
That sounds sensible. We can always move things around later if we want to use more than one repo for example...

**Comment by dougajmcdonald** [comment]
Yeah, I was thinking about multiple repos:

pros:
- deploy independently
- version independently
- smaller individual codebases
- can probably use git submodules to allow changes from other repos to be brought down easily
- possible to push to npm and use as dependencies that way in eventual releases etc

cons:
- duplicated infrastructure, npm, webpack configs etc which could diverge
- unsure whether even with git submodules whether multiple branches, pushes, PR's would be needed to update files across repo's requiring more of a context switch and hassle to make small updates

**Comment by gareththegeek** [comment]
Interesting, I think it's probably not a priority right now.  We should definitely focus on getting something that is simple and works first!

**Comment by dougajmcdonald** [comment]
Should be resolved with #10
